tests/kafkatest/services/kafka/kafka.py

```diff
@@ -280,7 +280,8 @@ def __init__(self, context, num_nodes, zk, security_protocol=SecurityConfig.PLAI
         self.configured_for_zk_migration = False
         
         # Set use_new_coordinator based on context and arguments.
-        default_use_new_coordinator = False
+        # The new group coordinator is used by default in kraft mode.
+        default_use_new_coordinator = self.quorum_info.using_kraft
```

Comment:
We can only use it in kraft mode. The issue is that we have logic in some tests which adds `consumer` to the enabled rebalance protocols when the new coordinator is enabled. This fails in zk mode because `consumer` is not supported.